import "CommonTypes.proto";

option py_generic_services = true;

message ActionRequest {
    required string capability = 1;
    required int64 time_out = 2;
    repeated KeyValueMessage params_list = 10;
}

message ActionResponse {
    optional string output = 6;
    optional string error_text = 7;
    optional int32 system_rc = 8;
    required string statusmsg = 3;
    required bool success = 9;
}

service ActionHandler_Service {
    rpc PerformAction(ActionRequest) returns (ActionResponse);
}
